database/generateData.js

```diff
@@ -1,55 +1,65 @@
 const fs = require('fs');
 const faker = require('faker');
 
-const write = function(filename, numRecords) {
-  return new Promise((resolve, reject) => {
+let name;
+let numRec;
+let i = 1;
```

Comment:
IMO this should be moved to line 37 above the while loop, as it is only used within the while loop

Reply:
I had to take them out of the write function to prevent re-initialization during recursion